File: c/src/common/vegetablesoup.c

#include "vegetablesoup.h"

soup_t * vegetablesoup_new()
{
	soup_t * soup = soup_new();
	soup->name = "Vegetable Soup";
	arraylist_string_add( soup->ingredients,"1 cup bullion");    
	arraylist_string_add( soup->ingredients,"1/4 cup carrots");         
	arraylist_string_add( soup->ingredients,"1/4 cup potatoes");         
   return soup;
}

File: go/creational/common/VegetableSoup.go
package common

type VegetableSoup struct {
	SoupImpl
}

func NewVegetableSoup() *VegetableSoup {
	return &VegetableSoup{ SoupImpl{ 
   "Vegetable Soup",
   []string{
	"1 cup bullion",
	"1/4 cup carrots",
	"1/4 cup potatoes",
	}}}
}
